DOCSP-34561 Geospatial section for RQL reference (#3167)
mongodb/docs-realm: source/realm-query-language.txt

Realm Query Language
====================

.. meta::

>   :description: Learn more about using Realm Query Language (RQL) for retrieving objects from a realm when using Atlas Device SDK.

Reviewer: Nit: I think we can simplify this meta description and omit some of the - in this context, confusing - product naming.

```suggestion
  :description: Learn how to use Realm Query Language (RQL) to retrieve objects from the database.
```

     - true
     - 0 and 1 are both less than none of 1 and 2

.. _rql-geospatial:

Geospatial Queries
------------------

You can query against geospatial data using the ``geoWithin`` operator.
The ``geoWithin`` operator takes the ``coordinates`` property of an embedded object 
as the coordinates of the point we're querying for and a geospatial shape. The operator checks whether 
the point is contained within the geospatial shape. 

In the following query, we are checking that the coordinates of the ``location`` point
are contained within the geospatial shape, ``smallCircle``:

> .. literalinclude:: /examples/generated/node/v12/geospatial.test.snippet.rql-geospatial.js

Reviewer: I think we need more setup for this example here. The rest of the examples on the page (I assume) work with the "[Examples on This Page](https://preview-mongodblindseymoore.gatsbyjs.io/realm/DOCSP-34561/realm-query-language/#examples-on-this-page)" section where we show the item and project models, etc. This example doesn't use those models, so it's not obvious what the `location` is here or what the `smallCircle` is.

We should probably expand this section more to describe the various geospatial shapes, talk about the shape that the coordinates data has to take, etc. You could draw inspiration from the new [Kotlin geospatial query section](https://www.mongodb.com/docs/realm/sdk/kotlin/realm-database/crud/read/#query-geospatial-data), or the upcoming [Swift geospatial docs](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-32521/sdk/swift/crud/read/#query-geospatial-data).

     - true
     - 0 and 1 are both less than none of 1 and 2

.. _rql-geospatial:

Geospatial Queries
------------------

You can query against geospatial data using the ``geoWithin`` operator.
The ``geoWithin`` operator takes the latitude/longitude pair in a custom 
embedded object's ``coordinates`` property and a geospatial shape. The 
operator checks whether the ``cordinates`` point is contained within the 
geospatial shape. 

The following geospatial shapes are supported for querying:

- ``GeoCircle``
- ``GeoBox``
- ``GeoPolygon``

To query geospatial data:

1. Create an object with a property containing the embedded geospatial data.
2. Define the geospatial shape to set the boundary for the query.
3. Query using the GEOWITHIN RQL operator. 

In the following query, we are checking that the coordinates of the embeddeded 
``location`` property are contained within the ``GeoCircle`` shape, ``smallCircle``:

.. literalinclude:: /examples/generated/node/v12/geospatial.test.snippet.rql-geospatial.js
     :language: js

For more information on defining geospatial shapes and objects with embedded geospatial data, 
see the geospatial documentation for your SDK:

> .. tabs::

Reviewer: We should switch this from using tabs to a bullet list. I merged the [Swift geospatial docs](https://www.mongodb.com/docs/realm/sdk/swift/crud/read/#query-geospatial-data) last week, so Swift now supports this. If we add Swift to the tab list, it will overflow the available width of the tab list and one or more tabs will be cut off.
